include/cuco/extent.cuh

```diff
@@ -125,43 +150,100 @@ template <typename Container, typename SizeType, std::size_t N>
  *
  * @return Resulting valid extent
  */
-template <typename Container, typename SizeType>
+template <int32_t CGSize, int32_t BucketSize, typename SizeType>
+[[nodiscard]] auto constexpr make_bucket_extent(SizeType size);
+
+/**
+ * @brief Computes valid bucket extent/capacity based on given parameters.
+ *
+ * @deprecated Use the equivalent `make_bucket_extent` instead.
+ *
+ * @tparam CGSize Number of elements handled per CG
+ * @tparam BucketSize Number of elements handled per Bucket
+ * @tparam SizeType Size type
+ *
+ * @param size The input size
+ *
+ * @throw If the input size is invalid
+ *
+ * @return Resulting valid extent
+ */
+template <int32_t CGSize, int32_t BucketSize, typename SizeType>
 [[nodiscard]] auto constexpr make_window_extent(SizeType size);
 
+template <typename ProbingScheme, typename Storage, typename SizeType, std::size_t N>
+[[nodiscard]] auto constexpr make_bucket_extent(cuco::extent<SizeType, N> ext);
+
+template <typename ProbingScheme, typename Storage, typename SizeType>
+[[nodiscard]] auto constexpr make_bucket_extent(SizeType ext);
+
 /**
- * @brief Computes valid window extent based on given parameters.
+ * @brief Computes a valid bucket extent/capacity for a given container type.
  *
  * @note The actual capacity of a container (map/set) should be exclusively determined by the return
  * value of this utility since the output depends on the requested low-bound size, the probing
  * scheme, and the storage. This utility is used internally during container constructions while for
  * container ref constructions, it would be users' responsibility to use this function to determine
- * the input size of the ref.
+ * the capacity ctor argument for the container.
  *
- * @tparam CGSize Number of elements handled per CG
- * @tparam WindowSize Number of elements handled per Window
+ * @tparam Container Container type to compute the extent for
  * @tparam SizeType Size type
  * @tparam N Extent
  *
  * @param ext The input extent
  *
  * @throw If the input extent is invalid
  *
- * @return Resulting valid extent
+ * @return Resulting valid `bucket extent`
+ */
+template <typename Container, typename SizeType, std::size_t N>
+[[nodiscard]] auto constexpr make_bucket_extent(extent<SizeType, N> ext);
+
+/**
+ * @brief Computes a valid bucket extent/capacity for a given container type.
+ *
+ * @deprecated Use the equivalent `make_bucket_extent` instead.
+ *
+ * @tparam Container Container type to compute the extent for
+ * @tparam SizeType Size type
+ * @tparam N Extent
+ *
+ * @param ext The input extent
+ *
+ * @throw If the input extent is invalid
+ *
+ * @return Resulting valid `bucket extent`
  */
-template <int32_t CGSize, int32_t WindowSize, typename SizeType, std::size_t N>
+template <typename Container, typename SizeType, std::size_t N>
 [[nodiscard]] auto constexpr make_window_extent(extent<SizeType, N> ext);
```

Comment:
Do we keep this for now for backwards compatibility in cudf?

Reply:
yes